test/no-array-instanceof.js

```diff
@@ -31,38 +31,23 @@ ruleTester.run('avoid-array-instanceof', rule, {
 			errors
 		},
 		{
-			code: 'arr instanceof Array === true',
-			output: 'Array.isArray(arr) === true',
-			errors
-		},
-		{
 			code: '[] instanceof Array',
 			output: 'Array.isArray([])',
 			errors
 		},
 		{
-			code: '[] instanceof Array === true',
-			output: 'Array.isArray([]) === true',
-			errors
-		},
-		{
-			code: '[1,2,3] instanceof Array',
-			output: 'Array.isArray([1,2,3])',
-			errors
-		},
-		{
-			code: '[1,2,3] instanceof Array === true',
-			output: 'Array.isArray([1,2,3]) === true',
+			code: 'fun.call(1, 2, 3) instanceof Array',
+			output: 'Array.isArray(fun.call(1, 2, 3))',
 			errors
 		},
 		{
-			code: 'if(arr instanceof Array){}',
-			output: 'if(Array.isArray(arr)){}',
+			code: 'obj.arr instanceof Array',
+			output: 'Array.isArray(obj.arr)',
 			errors
 		},
 		{
-			code: 'if([1,2,3] instanceof Array){}',
-			output: 'if(Array.isArray([1,2,3])){}',
+			code: `foo.bar${value} instanceof Array`,
```

Comment:
> What I wanted to see was if the instanceof operator was part of a string, what would happen? (Sorry, i am just curious, I have no experience with AST's ).

And that is a very good question to ask yourself :)
If you're not well versed with AST, then you are bound to get surprises (and even if you are :sweat_smile:)

I asked myself the same question, but could not find a test case that would make much sense (`arr "instanceof" Array`?) . Feel free to add a `"arr instanceof Array"` though!